### ISSUE TITLE
Error when using Requires.private with version constraint

### ISSUE THREAD

**Issue by fpoirotte** [pkgconf/pkgconf#359]
# Summary

When a .pc file contains Requires dependencies making use of a version constraint, parsing of the dependency fails if the constraint operator is not preceded and followed by a space.

I stumbled across this issue in our CI where .pc files are sometimes generated automatically and lack some of the spaces.

# Environment

Tested in a container with a fresh build of pkgconf 2.2.0.

# Steps to reproduce

I have 2 .pc files, visible below:
```
# ls -l *.pc
-rw-r--r--. 1 root root 304 Jun 28 17:05 check.pc
-rw-r--r--. 1 root root 232 Jun 28 17:02 libsubunit.pc
```

`libsubunit.pc` contains the following:
```
# cat libsubunit.pc
prefix=/usr
exec_prefix=/usr
libdir=/usr/lib64
includedir=/usr/include

Name: subunit
Description: Subunit test protocol library.
URL: http://launchpad.net/subunit
Version: 1.4.2
Libs: -L${libdir} -lsubunit
Cflags: -I${includedir}
```

`check.pc` contains the following:
```
# cat check.pc
prefix=/usr
exec_prefix=${prefix}
libdir=/usr/lib64
includedir=${prefix}/include

Name: Check
Description: A unit test framework for C
URL: https://libcheck.github.io/check/
Version: 0.15.2

Requires.private: libsubunit >=0.9.0

Libs: -L${libdir} -lcheck
Libs.private:   -lm -lrt
Cflags: -I${includedir}
```

Notice the `Requires.private` line that adds a versioned dependency on libsubunit and lacks a space after the operator.

Try to list flags for the "check" module:
```
# PKG_CONFIG_PATH=$(pwd) ./pkgconf --short-errors --print-errors --cflags --libs check
Package dependency requirement 'libsubunit >= .9.0' could not be satisfied.
Package 'libsubunit' has version '1.4.2', required version is '>= .9.0'
```

Add a space after the equal sign and remove the space before the ">" in check.pc.
This leads to a different error:
```
# PKG_CONFIG_PATH=$(pwd) ./pkgconf --short-errors --print-errors --cflags --libs check
Package 'libsubunit>=', required by 'check', not found
Package '0.9.0', required by 'check', not found
```

When there is a space before and after the ">=" operator, this works as expected:
```
# PKG_CONFIG_PATH=$(pwd) ./pkgconf --short-errors --print-errors --cflags --libs check
-I/usr/include -L/usr/lib64 -lcheck
```

Strangely enough, changing the requirement from ">=0.9.0" to e.g. ">=1.1.0" (with a space before the operator, but not after) also works correctly!
```
# PKG_CONFIG_PATH=$(pwd) ./pkgconf --short-errors --print-errors --cflags --libs check
-I/usr/include -L/usr/lib64 -lcheck
```

The behavior seems inconsistent. Is this expected?

**Comment by dg0yt** [comment]
> Strangely enough, changing the requirement from ">=0.9.0" to e.g. ">=1.1.0" (with a space before the operator, but not after) also works correctly!

Special treatment of leading zeros... `>=00.9.0` works, too.

Test with existing test data:
~~~
$ ./pkgconf --with-path=$SRCDIR/tests/lib1 --modversion "foo >0.5" 
Package dependency requirement 'foo > .5' could not be satisfied.
Package 'foo' has version '1.2.3', required version is '> .5'
~~~

**Comment by dg0yt** [comment]
> Special treatment of leading zeros... `>=00.9.0` works, too.

Actually the character which follows the operator is completely ignored.

**Comment by kaniini** [comment]
Fixed as of 8b7e2c4eee6ed0214626f665ea22d9ef8b4a4dd0
